### DISCUSSION TITLE
rutorrent unable to connect

### DISCUSSION THREAD

**Discussion by tryoutuser** [Novik/ruTorrent#2382] [Q&A]
hello.
i need help to check and correct me where i did wrong.

my vps is debian 11 with kernel 5.10.0-13-amd64.
i'm using nginx from nginx-extras package.
no ufw no firewalls.
rtorrent version 0.9.8 installed using apt.
rutorrent taken from releases 3.10.

however i get white page like rutorrent unable to connect :
![image](https://user-images.githubusercontent.com/35561383/189615593-58099656-d547-4fb1-a481-28bc922d557a.png)

installed php : https://pastebin.com/c5aUH4uX

rutorrent config.php : https://pastebin.com/7xguCQrr

my nginx config : https://pastebin.com/w3vDa25e

my rtorrent.rc : https://pastebin.com/BJm4Z6dZ

can somebody tell me what or where i set something wrong? thanks in advance!

**Comment by stickz** [comment]
hi @tryoutuser, could you post your php config file? There should be a setting to enable php error logs.

**Reply by tryoutuser** [reply]
the php.ini file or rutorrent php config?

**Reply by tryoutuser** [reply]
i'm sorry. i have checked and by default the `error_log` string is commented. but i have the error printed through nginx error.log.
should i post the error from nginx instead?

**Reply by stickz** [reply]
Yes, please do post any errors. Also uncomment the `error_log` string.

**Reply by tryoutuser** [reply]
here is the error :
```
[13-Sep-2022 22:12:10 Asia/Jakarta] PHP Warning:  Cannot load module 'http' because required module 'raphf' is not loaded in Unknown on line 0
```

**Reply by stickz** [reply]
This is a warning, not an error. It will not cause this problem to happen. 

Please uncomment the `error_log` string in your `php.ini` file. Afterwards, look for the newly generated php error log file.

**Reply by tryoutuser** [reply]
yes i did and that is what inside the generated `error_log = /var/log/php_errors.log` contained.

**Reply by stickz** [reply]
The default value for `log_errors` in `php.ini` file is `Off`. Change this to `On`. Afterwards, look again in this file for more errors.

**Reply by tryoutuser** [reply]
here is error from nginx : 
```
2022/09/13 22:15:37 [error] 492719#492719: *38873 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Class 'FileUtil' not found in /var/www/rutorrent-3.10/plugins/loginmgr/accounts.php:6
Stack trace:
#0 /var/www/rutorrent-3.10/plugins/loginmgr/init.php(5): require_once()
#1 /var/www/rutorrent-3.10/php/getplugins.php(495): require_once('/var/www/rutorr...')
#2 {main}  thrown in /var/www/rutorrent-3.10/plugins/loginmgr/accounts.php on line 6" while reading response header from upstream, client: ipaddress, server: domain, request: "GET /php/getplugins.php HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php7.4-fpm.sock:", host: "domain", referrer: "https://domain/"
2022/09/13 22:20:42 [error] 492719#492719: *39006 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Class 'FileUtil' not found in /var/www/rutorrent-3.10/plugins/loginmgr/accounts.php:6
Stack trace:
#0 /var/www/rutorrent-3.10/plugins/loginmgr/init.php(5): require_once()
#1 /var/www/rutorrent-3.10/php/getplugins.php(495): require_once('/var/www/rutorr...')
#2 {main}  thrown in /var/www/rutorrent-3.10/plugins/loginmgr/accounts.php on line 6" while reading response header from upstream, client: ipaddress, server: domain, request: "GET /php/getplugins.php HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php7.4-fpm.sock:", host: "domain", referrer: "https://domain/"
2022/09/13 22:25:10 [error] 492719#492719: *39101 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Class 'FileUtil' not found in /var/www/rutorrent-3.10/plugins/loginmgr/accounts.php:6
Stack trace:
#0 /var/www/rutorrent-3.10/plugins/loginmgr/init.php(5): require_once()
#1 /var/www/rutorrent-3.10/php/getplugins.php(495): require_once('/var/www/rutorr...')
#2 {main}  thrown in /var/www/rutorrent-3.10/plugins/loginmgr/accounts.php on line 6" while reading response header from upstream, client: ipaddress, server: domain, request: "GET /php/getplugins.php HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php7.4-fpm.sock:", host: "domain", referrer: "https://domain/"
```

**Reply by tryoutuser** [reply]
it was already `On` by default after installing via apt : 
![image](https://user-images.githubusercontent.com/35561383/189943861-30860c0d-f24d-4b5a-b63a-4669f0dfa252.png)

**Reply by stickz** [reply]
This is the log information we need. rutorrent version 3 and 4 are not backwards compatible. You can not run v4 plugins on v3.10.

The best way to resolve this problem is to upgrade rutorrent to version 4. It is recommended to backup your entire rutorrent folder. Then copy and replace absolutely everything. Afterwards, clear your web browser cache and refresh the page. 

This commit is recommended due to important fixes. [55e533a](https://github.com/Novik/ruTorrent/archive/55e533ae4b7841fe10beaec04048d3ada698075f.zip) It is very stable. The only major change is with the RSS plugin. In the unlikely event you experience a problem, you can disable the RSS plugin and create an issue report on GitHub.

**Reply by tryoutuser** [reply]
is this commit fine? i'm not sure if i use the beta version or this one, i usually tend to follow a stable release.

about rss plugin i never use it so it's fine i can disable it.

**Reply by tryoutuser** [reply]
is this correct? double up parent dir won't that be outside of rutorrent folder?
![image](https://user-images.githubusercontent.com/35561383/190125051-ed862f43-c272-43fa-bf17-4b282b61c4b8.png)

**Reply by tryoutuser** [reply]
ok i corrected that `$profilePath` and it runs.
but i've got these :
```
[15.09.2022 20:03:03] _cloudflare: Plugin will not work. Webserver user can't access external program (python).
[15.09.2022 20:03:03] spectrogram: Plugin will not work. rTorrent user can't access external program (sox).
```
what version of python does it need? i have python2.7 (2.7.18) and python3.9 (3.9.2) installed.
sox i can install it later.

**Reply by tryoutuser** [reply]
also some themes are not working, such as dark better and material design.

**Reply by stickz** [reply]
Hi @tryoutuser, the commit I sent you is the most stable. The only reason I haven't created a new release yet, is because we are still testing a major change to the RSS plugin. 

The `$profilePath` is correct as shown in the new release. It has to go back two directories to reach the parent directory. It starts in the newly created utility folder. It must be set like this, otherwise it will create the profile path in the php folder and cause other problems.

If python is installed, make sure it's accessible by the web server. What is the output of `command -v python` ?

The reason these two themes are not working properly is because they link to a third party repository. It's problematic when people download the latest zip from our releases. They do not install properly. I will fix this issue. You can get them here and install manually.
https://github.com/Novik/ruTorrent/tree/master/plugins/theme/themes

**Reply by tryoutuser** [reply]
![image](https://user-images.githubusercontent.com/35561383/190648904-5542e4e6-86bd-45dd-a38d-3c06452ab4b7.png)

weird i did not see `utility` folder. it goes to `share` folder instead.

`command -v python` returned nothing.

**Reply by stickz** [reply]
The `utility` folder is inside the `php` folder. The `$profilePath` variable goes `utility->php->ruTorrent`. It is not recommended to change the variable like this. The `share` folder is not intended create and execute inside your `php` folder. If the new profile path doesn't work, it's better to fix the root cause of the problem, than to create anther problem.

`python` is not working by default because Debian 11 removed the symlink when deprecating version 2. Python 2 is almost obsolete. The best course of action is test Python 3 for legacy scripts. The old version may stop working in the near future.

You can do this with `sudo apt install python-is-python3`. However, it's important to be mindful of any issues. Compatibility for legacy scripts is not guaranteed. It's good to report any problems quickly, to increase the chances of a script update.

**Reply by tryoutuser** [reply]
ok i reverted the profile path back to `../../share` i thought it's still the same like old version which is just one parent dir up `../share`

if enabling python symlink is possibly causing issues then i rather don't do it. i'll disable cloudflare plugin.

**Comment by Oreloth** [comment]
It seems to be a permission issue, I had same error on Debian before I switch to EL8. Are rtorrent and ruTorrent running with the same user?

**Reply by tryoutuser** [reply]
yes, everything is under the same user. i have rtorrent running with screen and user+group permissions are allowed to read, write and execute.

**Reply by Oreloth** [reply]
Can you try to remove the most recent folder in the following path?

/var/www/rutorrent-3.10-1/share/users/<username>/settings/tasks

Sometimes this folder can cause issues while a task was not closed properly.
